client/packages/programs/src/JsonForms/common/components/Array/FirstItemArray.tsx

```diff
@@ -88,19 +88,13 @@ const FirstItemArrayComponent = (props: FirstItemArrayControlCustomProps) => {
         </Box>
       ) : null}
 
-      <JsonForms
+      <JsonFormsDispatch
         key={childPath}
         schema={schemaWithDefs}
         uischema={childUiSchema || uischema}
-        data={data?.[0] ?? {}}
-        config={config}
-        readonly={!enabled}
-        renderers={renderers ?? []}
-        onChange={({ data }) => {
-          if (Object.values(data).filter(it => !!it).length > 0) {
-            handleChange(childPath, data);
```

Comment:
This was the culprit -- Falsy values were being filtered out so the "handleChange" function didn't run